runtime/src/iouring/mod.rs

```diff
@@ -2,10 +2,16 @@ use futures::{
     channel::{mpsc, oneshot},
     StreamExt as _,
 };
-use io_uring::{opcode::LinkTimeout, squeue::Entry as SqueueEntry, types::Timespec, IoUring};
-use std::time::Duration;
+use io_uring::{
+    opcode::{LinkTimeout, Timeout},
+    squeue::Entry as SqueueEntry,
+    types::Timespec,
+    IoUring,
+};
+use std::{collections::HashMap, time::Duration};
 
 const TIMEOUT_WORK_ID: u64 = u64::MAX;
+const SHUTDOWN_TIMEOUT_WORK_ID: u64 = u64::MAX - 1;
```

Comment:
Might be worth it to use an enum at this point to make the code more robust?

```
enum Work {
  ID(u64),
  Timeout
  ShutdownTimeout,
};
```

Reply:
Unfortunately, they all have to be `u64` because we tag each submitted operation using `user_data(u64)`.